annotation-utils/src/main/java/edu/isi/vista/annotationutils/PushAnnotations.kt

```diff
@@ -0,0 +1,145 @@
+                    addIdentity(sshKeyPath)
+
+                }
+            }
+        }
+    }
+
+    val git = Git.open(File(repoPath))
+    val config = git.getRepository().getConfig()
+    config.setString(
+            "remote",
+            "origin",
+            "url",
+            "git@github.com:isi-vista/curated-training-annotation.git")
+    config.save()
+
+    val currentDate = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
+    TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
+    val formatter = SimpleDateFormat("yyyy-MM-dd-HH-mm-ss")
+
+    // Currently working on test_branch
+    git.checkout().setName("test_branch").call()
+
+    git.add().addFilepattern("data/").call()
+    val newCommit = git.commit()
+            .setMessage("includes annotations from ${currentDate.getTime()}").call()
+    git.tag().setName(formatter.format(currentDate.getTime())).call()
+
+
+    if (hasDiff(git, newCommit)) {
+        val pushIter = git.push()
+                .setTransportConfigCallback(SshConfig("~/.ssh/id_rsa"))
+                .call()
+        val pushResult = pushIter.iterator().next()
+    } else {
+        println("No new annotations - nothing will be pushed.")
+        git.reset().setRef("HEAD~1").setMode(ResetCommand.ResetType.SOFT).call()
+    }
+}
+
+// Compare old data to newly exported data
+fun hasDiff(git: Git, newCommit: RevCommit): Boolean {
+    val oldCommit = getPrevCommit(git, newCommit)
+    if (oldCommit == null) {
+        println("This is the first commit.")
+        return true
+    }
+    val oldTree = getCanonicalTreeParser(git, oldCommit)
+    val newTree = getCanonicalTreeParser(git, newCommit)
+    val diffs: List<DiffEntry> =
+            git.diff().setNewTree(newTree).setOldTree(oldTree).call()
+    if (diffs.isEmpty()) {
+        return false
+    }
+    return true
+}
+
+fun getPrevCommit(git: Git, commit: RevCommit): RevCommit? {
+    val walk = RevWalk(git.getRepository())
+    walk.markStart(commit)
+    var count = 0
+    for (rev in walk) {
+        if (count ==1) {
+            return rev
+        }
+        count = count.inc()
+    }
+    return null
+}
+
+fun getCanonicalTreeParser(git: Git, commit: ObjectId): CanonicalTreeParser {
+    val walk = RevWalk(git.getRepository())
+    val revCommit = walk.parseCommit(commit)
+    val treeId = revCommit.getTree().getId()
+    val reader = git.getRepository().newObjectReader()
+    return CanonicalTreeParser(null, reader, treeId)
+}
+
+
+fun main(argv: Array<String>) {
+    val paramsLoader = SerifStyleParameterFileLoader.Builder()
+            .interpolateEnvironmentalVariables(true).build()
+    val params = paramsLoader.load(File(argv[0]))
+    val exportAnnotationsParams = params.getString("exportAnnotationsParams")
+    val restoreJsonParams = params.getString("restoreJsonParams")
+    val repoPath = params.getString("repoPath")
+
+    // Run ExportAnnotations.kt
+    ExportAnnotations.main(arrayOf(exportAnnotationsParams))
```

Comment:
We should modify these parameters to put the correct directories in so the user doesn't need to coordinate it by hand. The easiest way to do this is to rename the current main methods of these to something which takes a `Parameters` object as its argument.  Then add `main` methods to them which loads the parameters and calls that method. That way we can call the parameters-as-args methods with modified parameters objects from this script.

Reply:
Sorry for the confusion - how does this prevent the user from needing to put the correct directory by hand?

Reply:
So, as it is currently, the user needs to coordinate certain paths to "match up" between the three parameter files. We would rather generate those parameters automatically so that the user doesn't have to worry about it.

Reply:
(and you never have to apologize for being confused. It's my job as a reviewer (and more senior developer) to be clear and your job to keep asking me questions until I manage to do so :-) )

Reply:
How would I automatically generate the parameters?

Reply:
So in `ExportAnnotations` for example, if I create a method out of `main` (call it `export`) and it takes a `Parameters` object as an argument, and then I call that method from `PushAnnotations`, where would the required `Parameters` come from if not specified by the user? I feel like I'm missing something here.

Reply:
Done